Lib/compileall.py

```diff
@@ -165,7 +164,7 @@ def compile_file(fullname, ddir=None, force=False, rx=None, quiet=0,
 
     if stripdir is not None:
         fullname_parts = fullname.split(os.path.sep)
-        stripdir_parts = stripdir.split(os.path.sep)
+        stripdir_parts = os.fspath(stripdir).split(os.path.sep)
```

Comment:
ISTM it would be clearer to put the input normalization (`os.fspath()`) up above with the normalization of `fullname` and `name`. I.e., add up above:

```python3
stripdir = os.fspath(stripdir) if stripdir is not None else None
```
```suggestion
        stripdir_parts = stripdir.split(os.path.sep)
```